policy-controller/k8s/index/src/meshtls_authentication.rs

```diff
@@ -29,6 +29,9 @@ impl Spec {
                 let ns = tgt.namespace.as_deref().unwrap_or(&namespace);
                 let id = cluster.service_account_identity(ns, &tgt.name);
                 Ok(IdentityMatch::Exact(id))
+            } else if tgt.targets_kind::<Namespace>() {
+                let id = cluster.namespace_identity(tgt.name.as_str());
+                Ok(id.parse::<IdentityMatch>()?)
```

Comment:
since this returns a result no need for shortcircuiting

```suggestion
                id.parse::<IdentityMatch>()
```